### PR TITLE
Added SIGNL4 as a notification option

### PR DESCRIPTION
**Pull request by rons4**
There is already a SIGNL4 package available here:
https://exchange.checkmk.com/p/signl4
As discussed I tried my best to incorporate that into this repository.
If you should have any questions please just let me know.

### CODE REVIEW

**Comment by jherbel** [comment]
Hi Ron,

thanks for your pull request. Before we can look at it more closely, all the tests need to work. See https://github.com/tribe29/checkmk/blob/master/CONTRIBUTING.md#how-to-execute-tests for information on how to run them locally.

Best
Jörg

**Comment by rons4** [comment]
Thanks Jörg,
I am working on it and I hope all checks will be fine soon.
Best Regards
Ron

**Comment by jherbel** [comment]
Hello Ron,
we will have a closer look at your patch and discuss internally on how and if we can implement it into the official code base.
Internal Ref: CMK-4908

**Comment by rons4** [comment]
Thanks a lot Jörg,

This is the main file from our notification package: https://exchange.checkmk.com/p/signl4.

I tries to add the configuration part under "\cmk\gui\plugins\wato\notifications.py" too:

```
@notification_parameter_registry.register
class NotificationParameterSIGNL4(NotificationParameter):
    @property
    def ident(self):
        return "signl4"

    @property
    def spec(self):
        return Dictionary(
            title=_("Create notification with the following parameters"),
            optional_keys = ['originator'],
            elements = [
                ("password", TextAscii(
                    title = _("Team Secret"),
                    help = _("The team secret of your SIGNL4 team. That is the last part of your webhook URL: https://connect.signl4.com/webhook/<team_secret>"),
                    size = 40,
                    allow_empty = False,
                )),
            ],
        )
```

And this here under "\notifications\signl4":
```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# Team alerting (using SIGNL4)

import sys

from cmk.notification_plugins import signl4

if __name__ == "__main__":
    sys.exit(signl4.main())

```

But this caused test errors.

The configuration is simple and just takes one parameter:

config

If you should have any questions please just let me know.

**Comment by jherbel** [comment]
Hi Ron,
thanks for the extra information. The developer who will take a closer look at the PR will come back to you in case we need more information or have any additional questions.
Best
Jörg

**Comment by rons4** [comment]
OK and thanks
Ron

File: cmk/gui/plugins/wato/notifications.py [hunk 1]
```diff
@@ -396,6 +396,28 @@ def spec(self):
...
+        return "signl4"
+
+    @property
+    def spec(self):
+        return Dictionary(
+            title=_("Create notification with the following parameters"),
+            optional_keys = [],
+            elements = [
+                ("password", Password(
+                    title = _("Team Secret"),
```
Comment (godspeed-you) [inline cmk/gui/plugins/wato/notifications.py:412]: You could alternatively implement a cascading dropdown to give the user a choice if he/she wants to write the team key directly or use a defined from a password store. As an example you can take a look at the implementation of slack.

File: tests-py3/unit/cmk/gui/wato/test_notification_parameters.py [hunk 2]
```diff
@@ -20,6 +20,7 @@
     'pagerduty',
     'pushover',
     'servicenow',
+    'signk4',
```
Comment (godspeed-you) [inline tests-py3/unit/cmk/gui/wato/test_notification_parameters.py:23]: small typo at this place (which leads to a failed test).

**Review by godspeed-you — COMMENTED** [review]
As discussed my notes to this PR.

**Comment by godspeed-you** [comment]
As discussed, we implemented a corrected version in the master.
Commit: e2a229520ee9e9c25d17740ddb894fe55d6c5cda

**Comment by rons4** [comment]
@godspeed-you, thanks a lot for all your help.

**Pull request closed without merging**